Replaced Synchronous Advanced Number Insight with Asynchronous Advanced Number Insight

GregHolmes opened the pull request:
Because the synchronous advanced number insight has been deprecated and is unreliable, I've replaced this functionality with the asynchronous advanced number insights. Which requires another endpoint and a place to store the returned response from the webhook.

lornajane reviewed (changes requested):
Looks good (I fixed a few things)! The advanced one does put JSON into a file, but then appends with the next request which might not be what we want. Also the template output says to "check the logs" but there's nothing in the logs, should we either link to the file in the user-facing output, or log what arrives? Either would be OK!

GregHolmes commented:
Thank you @lornajane . I've removed the appending functionality when saving to the file.
Updated the text to link to that file.
And I've also added a check to make sure a variable isset before checking whether it's true or false.

lornajane reviewed (approved):
This is such a huge improvement! Thanks Greg :+1:
